src/main/java/bio/terra/service/snapshot/flight/create/CreateSamGroupStep.java

```diff
@@ -0,0 +1,37 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.model.SnapshotRequestModel;
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.service.snapshot.flight.SnapshotWorkingMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+
+public class CreateSamGroupStep implements Step {
+  private final IamService iamClient;
+  private final SnapshotRequestModel snapshotReq;
+
+  public CreateSamGroupStep(IamService iamClient, SnapshotRequestModel snapshotReq) {
+    this.iamClient = iamClient;
+    this.snapshotReq = snapshotReq;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    context
+        .getWorkingMap()
+        .put(SnapshotWorkingMapKeys.SAM_GROUP, iamClient.createGroup(snapshotReq.getName()));
```

Comment:
"snapshotReq.getName()" - So, this would be the snapshot name (as opposed to the access request name set in the cohort builder ui), right? 
We've talked about moving naming snapshots for byRequestId into the create snapshot flight as opposed to passing it in through the UI. We should consider what that means here.

Reply:
Also, do you know what the requirements are for group names? If they don't match our snapshot name requirements, we'll need to do something different.

Reply:
I decided to follow the conventions the CreateDuosFirecloudGroupStep uses